Tests/SentryProfilerTests/SentryProfiledTracerConcurrencyTests.swift

```diff
@@ -0,0 +1,245 @@
+        XCTAssertTrue(SentryContinuousProfiler.isCurrentlyProfiling())
+
+        // Act
+        sentry_stopAndDiscardLaunchProfileTracer()
+
+        // Assert
+        XCTAssertEqual(_gInFlightRootSpans, 0)
+
+        // Act
+        try fixture.allowContinuousProfilerToStop()
+
+        // Assert
+        XCTAssertFalse(SentryContinuousProfiler.isCurrentlyProfiling())
+    }
+
+    func testLaunchProfileStartWithAdditionalSpan() throws {
+        // Arrange
+        fixture.options.profilesSampleRate = nil
+        fixture.options.configureProfiling = {
+            $0.lifecycle = .trace
+            $0.sessionSampleRate = 1
+            $0.profileAppStarts = true
+        }
+        fixture.givenSdkWithHub()
+
+        // Assert
+        XCTAssertEqual(_gInFlightRootSpans, 0)
+        XCTAssertFalse(SentryContinuousProfiler.isCurrentlyProfiling())
+
+        // Act
+        _sentry_nondeduplicated_startLaunchProfile()
+
+        // Assert
+        XCTAssertEqual(_gInFlightRootSpans, 1)
+        XCTAssertTrue(SentryContinuousProfiler.isCurrentlyProfiling())
+
+        // Act
+        let span = try fixture.newTransaction(rootSpan: true)
+
+        // Assert
+        XCTAssertEqual(_gInFlightRootSpans, 2)
+        XCTAssertTrue(SentryContinuousProfiler.isCurrentlyProfiling())
+        
+        // Act
+        span.finish()
+
+        // Assert
+        XCTAssertEqual(_gInFlightRootSpans, 1)
+        XCTAssertTrue(SentryContinuousProfiler.isCurrentlyProfiling()) 
+        
+        // Act
+        XCTAssertNotNil(sentry_launchTracer)
+        sentry_stopAndDiscardLaunchProfileTracer()
+
+        // Assert
+        XCTAssertEqual(_gInFlightRootSpans, 0)
+
+        // Act
+        try fixture.allowContinuousProfilerToStop()
+
+        // Assert
+        XCTAssertFalse(SentryContinuousProfiler.isCurrentlyProfiling())
+    }
+
+    class ProfiledViewController: UIViewController {}
+
+    func testLaunchProfileStartAndStopWithTTFD() throws {
+        // Arrange
+        fixture.options.profilesSampleRate = nil
+        fixture.options.configureProfiling = {
+            $0.lifecycle = .trace
+            $0.sessionSampleRate = 1
+            $0.profileAppStarts = true
+        }
+        fixture.options.enableTimeToFullDisplayTracing = true
+        fixture.options.add(inAppInclude: "uikitcore")
+        SentryDependencyContainer.sharedInstance()
+        SentrySDK.start(options: fixture.options)
+
+        // Assert
+        XCTAssertEqual(_gInFlightRootSpans, 0)
+        XCTAssertFalse(SentryContinuousProfiler.isCurrentlyProfiling())
+
+        // Act
+        _sentry_nondeduplicated_startLaunchProfile()
+
+        // Assert
+        XCTAssertEqual(_gInFlightRootSpans, 1)
+        XCTAssertTrue(SentryContinuousProfiler.isCurrentlyProfiling())
+
+        let uivc = ProfiledViewController(nibName: nil, bundle: nil)
+        uivc.loadView()
+        uivc.viewWillAppear(false)
+        fixture.displayLinkWrapper.call()
```

Comment:
`m`: As you pointed out on Slack, you want to have input on that test. I think we could make our lives easier here by limiting the setup and dependencies we require for testing. Ideally, yes we test with the started SDK, an UIViewController to start and finish a trace, but if this causes trouble, we can also emulate what that code is doing by directly interacting with the code we want to test. 
If the code we want to test requires all the setup, we could try to refactor the code so it is easier to test. I'm sorry it's a bit hard to explain in GH comment what exactly would be required here.

Reply:
Yes, I tried initially starting with just the components I had in mind, things kept failing in weird ways due to the tightly coupled nature of many parts of the SDK.

Reply:
Maybe we can refactor parts of the code we want to test so it depends on less code.